src/go/configgenerator/route_generator.go

```diff
@@ -271,18 +283,42 @@ func makeRouteTable(serviceInfo *configinfo.ServiceInfo) ([]*routepb.Route, erro
 			respTimeout = method.BackendInfo.Deadline
 		}
 
-		var routeMatchers []*routepb.RouteMatch
+		// The `methodNotAllowedRouteMatchers` are the route matches covers all the defined uri templates
+		// but no specific methods. As all the defined requests are matched by `routeMatchers`, the rest
+		// matched by `methodNotAllowedRouteMatchers` fall in the category of `405 Method Not Allowed`.
+		var routeMatchers, methodNotAllowedRouteMatchers []*routepb.RouteMatch
```

Comment:
This code seems too complicated.   I feel it will be more cleaner if we can do followings
1) use a separate pass to loop through httpPatternsMethods, to get a map of path to bool.  If bool is true,  use regex, otherwise it is exact path match
2) then use that map to generate all 405 route matches

Reply:
IMHO, I feel it is better to put all the making routeMatch logic in one place, since the `methodNotAllowedRouteMatchers` are basically associated with `routeMatchers`.  It will be less easy to introduce bug later when we add more logic for routeMatchers.